recipe/meta.yaml

```diff
@@ -18,12 +22,18 @@
   {% set cuda_minor = 0 %}
 {% endif %}
 
+# encode mpi dependency in the build string
+# and prioritize nompi variant via build number
 {% if mpi != "nompi" %}
   {% set mpi_prefix = "mpi_" + mpi %}
-  {% set mpi_suffix = sdk_build + 100 %}
+  # TODO: next release, set increment to 0 instead of 100
+  # keep it for now to avoid generating new uploads
+  {% set mpi_suffix = sdk_build + 100 + build_num %}
 {% else %}
   {% set mpi_prefix = "nompi_" %}
-  {% set mpi_suffix = sdk_build %}
+  # TODO: set increment to 100 instead of 200
+  # setting the increment to 200 is a temporary fix to prioritize nompi build over mpi builds
+  {% set mpi_suffix = sdk_build + 200 + build_num %}
```

Comment:
You want to swap the priority of the MPI variants to make MPI higher priority than nompi? Why? Is it too difficult for users to just install both mpi and cuquantum at the same time?

Waiting to the next version is not necessary. Just swap the priority of the MPI and publish a new build. Adding a 200 build offset of not necessary because this release will have build numbers 101 and 1, so they are already higher than 100 and 0.

Reply:
The goal is to have the `nompi` build higher priority, which is not the case now.

Unfortunately the build string construction in the recipe is a bit inconsistent. Prior versions of the recipe never used a proper build number and instead used the build number from the release. The current version uses a proper build number for `cuquantum` output, but not for the other outputs: cutensornet, cudensitymat and custatevec all have build strings ending on the release build number (+increment for build prioritization). This should be fixed next release.